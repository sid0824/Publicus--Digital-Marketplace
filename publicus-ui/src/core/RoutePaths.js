// The paths of every page in the site
export default {
  Home: '/',
  FormCreator: '/formCreator',
  CreateAccount: '/createAccount',
  Login: '/login',
  Marketplace : '/marketplace',
  MyForms: '/myForms',
  CreateForm: '/createForm',
  ViewResults: '/viewResults',
  CompleteForm: '/completeForm',
  MyAccount: '/myAccount',
  Administration: '/administration'
}