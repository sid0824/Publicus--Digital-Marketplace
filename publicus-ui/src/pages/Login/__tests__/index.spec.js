import toJson from 'enzyme-to-json';
import React from "react";
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Login} from "../index";
import {Auth} from "aws-amplify";
import { message } from 'antd';
import RoutePaths from "../../../core/RoutePaths";
configure({ adapter: new Adapter() });

const mockValues = {
  email: 'EMAIL',
  password: 'PASSWORD'
};

const mockGetFieldDecorator = jest.fn();
const mockForm = {
  validateFields: jest.fn().mockImplementation(async (func) => func(null, mockValues)),
  getFieldDecorator: mockGetFieldDecorator,
};

const mockSetUserHasAuthenticated = jest.fn();
const mockPush = jest.fn();
const mockHistory = {
  push: mockPush
};

describe('Login', () => {

  beforeEach(() => {
    mockGetFieldDecorator.mockReturnValue((component) => component);
    Auth.signIn = jest.fn();
    message.success = jest.fn();
    message.error = jest.fn();
  });

  afterEach(() => {
    mockSetUserHasAuthenticated.mockReset();
    Auth.signIn.mockReset();
    message.success.mockReset();
    message.error.mockReset();
  });

  test('should render the login page', () => {
    const wrapper = shallow(
      <Login
        form={mockForm}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should login', async (done) => {
    const wrapper = shallow(
      <Login
        form={mockForm}
        history={mockHistory}
        setUserHasAuthenticated={mockSetUserHasAuthenticated}
      />
    );
    await wrapper.instance().onLogin({ preventDefault: () => {} });
    setTimeout(() => {
      expect(message.success).toHaveBeenCalledWith('Logged In Successfully');
      expect(mockSetUserHasAuthenticated).toHaveBeenCalledWith(true);
      expect(Auth.signIn.mock.calls[0]).toMatchSnapshot();
      expect(mockPush.mock.calls[0][0]).toEqual(RoutePaths.Home);
      done();
    }, 1100);
  });

  test('should give error on failing to login', async (done) => {
    const wrapper = shallow(
      <Login
        form={mockForm}
        history={mockHistory}
        setUserHasAuthenticated={mockSetUserHasAuthenticated}
      />
    );
    Auth.signIn.mockImplementation(() => {
      throw new Error("ERROR");
    });
    await wrapper.instance().onLogin({ preventDefault: () => {} });
    setTimeout(() => {
      expect(message.error).toHaveBeenCalledWith('ERROR');
      done();
    }, 1100);
  });

});