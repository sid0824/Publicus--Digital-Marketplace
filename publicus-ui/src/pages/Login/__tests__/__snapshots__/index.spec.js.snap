// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should login 1`] = `
Array [
  "EMAIL",
  "PASSWORD",
]
`;

exports[`Login should render the login page 1`] = `
<Form
  colon={true}
  hideRequiredMark={false}
  layout="horizontal"
  onSubmit={[Function]}
  style={
    Object {
      "marginLeft": "15%",
      "marginRight": "15%",
      "marginTop": "70px",
      "width": "70%",
    }
  }
>
  <h1>
    Enter Your Details
  </h1>
  Email Address
  <FormItem
    hasFeedback={false}
  >
    <Input
      placeholder="Username..."
      type="text"
    />
  </FormItem>
  Password
  <FormItem
    hasFeedback={false}
  >
    <Input
      placeholder="Password..."
      type="password"
    />
  </FormItem>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <FormItem
      hasFeedback={false}
    >
      <Button
        block={false}
        ghost={false}
        htmlType="submit"
        loading={false}
        type="primary"
      >
        Login
      </Button>
    </FormItem>
    <Link
      to="/createAccount"
    >
      <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
        style={
          Object {
            "marginLeft": 4,
            "marginTop": 3,
          }
        }
      >
        Create Account
      </Button>
    </Link>
  </div>
</Form>
`;
